app/models/manageiq/providers/openstack/cloud_manager.rb

```diff
@@ -403,6 +403,24 @@ def self.params_for_create
     }
   end
 
+  def self.create_from_params(params, endpoints, authentications)
+    if params.delete("event_stream_selection") == "ceilometer"
+      default_endpoint = endpoints.detect { |ep| ep["role"] == "default" }
+      endpoints << {"role" => "ceilometer", "hostname" => default_endpoint["hostname"]}
+    end
+
+    super
+  end
+
+  def edit_with_params(params, endpoints, authentications)
+    if params.delete("event_stream_selection") == "ceilometer"
+      default_endpoint = endpoints.detect { |ep| ep["role"] == "default" }
+      endpoints << {"role" => "ceilometer", "hostname" => default_endpoint["hostname"]}
+    end
```

Comment:
Are both of these methods the same code (aside from the super)?  If so, can we dry it up into a private method?

Reply:
:+1: yeah can definitely do that

Reply:
Done